speaker/main.go

```diff
@@ -280,6 +284,10 @@ func (c *controller) SetBalancer(l log.Logger, name string, svc *v1.Service, eps
 		return c.deleteBalancer(l, name, "notLoadBalancer")
 	}
 
+	if nodes.ConditionStatus(c.nodes[c.myNode], v1.NodeNetworkUnavailable) == v1.ConditionTrue {
```

Comment:
Is this necessary? If everything is correct we'll get here from shouldAnnounce

Reply:
done